serow_ros/src/serow_driver.cpp

```diff
@@ -91,85 +89,67 @@ class SerowDriver {
+                        static_cast<double>(joint_state_data_.header.stamp.nsec) * 1e-9;
+                    joint.position = joint_state_data_.position[i];
+                    joint_measurements[joint_state_data_.name[i]] = std::move(joint);
                 }
-                joint_state_data_.pop();
 
                 // Create the base imu measurement
-                const auto& imu_data = base_imu_data_.front();
-                const auto timestamp = imu_data.header.stamp;
+                const auto timestamp = base_imu_data_.header.stamp;
                 serow::ImuMeasurement imu_measurement{};
                 imu_measurement.timestamp =
-                    static_cast<double>(imu_data.header.stamp.sec) +
-                    static_cast<double>(imu_data.header.stamp.nsec) * 1e-9;
+                    static_cast<double>(base_imu_data_.header.stamp.sec) +
+                    static_cast<double>(base_imu_data_.header.stamp.nsec) * 1e-9;
                 imu_measurement.linear_acceleration =
-                    Eigen::Vector3d(imu_data.linear_acceleration.x, imu_data.linear_acceleration.y,
-                                    imu_data.linear_acceleration.z);
+                    Eigen::Vector3d(base_imu_data_.linear_acceleration.x, base_imu_data_.linear_acceleration.y,
+                                    base_imu_data_.linear_acceleration.z);
                 imu_measurement.angular_velocity =
-                    Eigen::Vector3d(imu_data.angular_velocity.x, imu_data.angular_velocity.y,
-                                    imu_data.angular_velocity.z);
-                base_imu_data_.pop();
+                    Eigen::Vector3d(base_imu_data_.angular_velocity.x, base_imu_data_.angular_velocity.y,
+                                    base_imu_data_.angular_velocity.z);
 
                 // Create the leg F/T measurement
                 std::map<std::string, serow::ForceTorqueMeasurement> ft_measurements;
-                for (auto& [key, value] : ft_data_) {
-                    if (value.size()) {
-                        serow::ForceTorqueMeasurement ft{};
-                        const auto& ft_data = value.front();
-                        ft.timestamp = static_cast<double>(ft_data.header.stamp.sec) +
-                                       static_cast<double>(ft_data.header.stamp.nsec) * 1e-9;
-                        ft.force = Eigen::Vector3d(ft_data.wrench.force.x, ft_data.wrench.force.y,
-                                                   ft_data.wrench.force.z);
-                        ft.torque = Eigen::Vector3d(ft_data.wrench.torque.x,
-                                                    ft_data.wrench.torque.y,
-                                                    ft_data.wrench.torque.z);
-                        ft_measurements[key] = std::move(ft);
-                        value.pop();
+                if (ft_data_.size() == num_feet_){
```

Comment:
a better practice here would be 
```
if (ft_data_.size() != num_feet_) {
  // Throw an error and safely exit since SEROW estimates would be inaccurate with inaccurate/inexistent contact data
}
```